### ISSUE TITLE
Using --project flag with reporter will cause Cypress to look for reporter package inside the project folder

### ISSUE THREAD

**Issue by Alfy102** [cypress-io/cypress#28257]
### Current behavior

Consider the following project structure:
![image](https://github.com/cypress-io/cypress/assets/8023501/aec0c95c-bfb4-4ed0-ac8d-1b0b349915e1)

In both project, the defineConfig has the reporter set to 'cypress-multi-reporter':
```
module.exports = defineConfig ({
.
.
.
reporter: 'cypress-multi-reporters',
.
.
.
})
```

the launcher.js uses `cypres.run()` to  run cypress headless mode.

Running launcher using:
` npm run project1Test -- feature=InitialTest browser=chrome`
will get this error:
```
Error loading the reporter: cypress-multi-reporters

We searched for the reporter in these paths:

 - C:\Users\[username]\Documents\Git\testProject\project1\cypress-multi-reporters
 - C:\Users\[username]\Documents\Git\testProject\project1\node_modules\cypress-multi-reporters

Learn more at https://on.cypress.io/reporters

Error: Cannot find module 'C:\Users\bvxt47\Documents\Git\testProject\project1\node_modules\cypress-multi-reporters'
Require stack:
- C:\Users\[username]\AppData\Local\Cypress\Cache\12.17.4\Cypress\resources\app\packages\server\lib\reporter.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:940:15)
    at n._resolveFilename (node:electron/js2c/browser_init:249:1105)
    at PackherdModuleLoader._tryResolveFilename (evalmachine.<anonymous>:1:787129)
    at PackherdModuleLoader._resolvePaths (evalmachine.<anonymous>:1:783919)
    at PackherdModuleLoader.tryLoad (evalmachine.<anonymous>:1:781962)
    at Function.<anonymous> (evalmachine.<anonymous>:1:792852)
    at d._load (<embedded>:4509:29304)
    at Module.require (node:internal/modules/cjs/loader:1012:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at S.loadReporter (<embedded>:4514:19480)
    at C.initializeReporter (<embedded>:4561:138210)
    at C.startWebsockets (<embedded>:4561:138514)
    at C.open (<embedded>:4561:136843)
    at async v.create (<embedded>:4561:149367)
    at async O (<embedded>:4561:226641)
    at async W (<embedded>:4561:239882)
{
  status: 'failed',
  failures: 1,
  message: 'Could not find Cypress test run results'
}
```

Basically Cypress will look for the reporter package in node_modules folder inside the project file.

### Desired behavior

What should happen is for Cypress to look for the reporter package in the node_modules from root directory.

A workaround to this issue to set the reporter input to the root path of the reporter package:
```
module.exports = defineConfig ({
.
.
.
reporter: '../node_modules/cypress-multi-reporters/index.js',
.
.
.
})
```

Then it is able to find the cypress-multi-reporter from the root node_modules.

### Test code to reproduce

Clone this repo to reproduce this issue:
https://github.com/Alfy102/cypress-test-project-structure/tree/master

### Cypress Version

13.4.0

### Node version

20.9.0

### Operating System

Win11

**Comment by mbaird4** [comment]
I'm also getting this issue
